src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -182,10 +187,148 @@ public void delete(Long examId) {
+     * @return return ExamScoresDTO with students, scores and exerciseGroups for exam
+     */
+    public ExamScoresDTO getExamScore(Long examId) {
+        Exam exam = examRepository.findForScoreCalculationById(examId).orElseThrow(() -> new EntityNotFoundException("Exam with id: \"" + examId + "\" does not exist"));
+
+        // Adding exam information to DTO
+        ExamScoresDTO scores = new ExamScoresDTO(exam.getId(), exam.getTitle(), exam.getMaxPoints());
+
+        // Adding exercise group information to DTO
+        for (ExerciseGroup exerciseGroup : exam.getExerciseGroups()) {
+            // Alert: This only works if all exercises in an exercise groups have the same number of maximum points
+            Double maximumNumberOfPoints = null;
+            if (!exerciseGroup.getExercises().isEmpty()) {
+                maximumNumberOfPoints = exerciseGroup.getExercises().iterator().next().getMaxScore();
+            }
+
+            List<String> containedExercises = new ArrayList<>();
+
+            for (Exercise exercise : exerciseGroup.getExercises()) {
+                containedExercises.add(exercise.getTitle().trim());
+            }
+
+            scores.exerciseGroups.add(new ExamScoresDTO.ExerciseGroup(exerciseGroup.getId(), exerciseGroup.getTitle(), maximumNumberOfPoints, containedExercises));
+        }
+
+        // Adding registered student information to DTO
+        for (User user : exam.getRegisteredUsers()) {
+            scores.studentResults.add(new ExamScoresDTO.StudentResult(user.getId(), user.getName(), user.getEmail(), user.getLogin(),
+                    user.getRegistrationNumber() != null ? user.getRegistrationNumber().trim() : ""));
+        }
+
+        List<StudentParticipation> studentParticipations = exam.getExerciseGroups().stream().map(ExerciseGroup::getExercises).flatMap(Collection::stream)
+                .map(Exercise::getStudentParticipations).flatMap(Collection::stream).collect(Collectors.toList());
+
+        // Adding student results information to DTO
+        for (ExamScoresDTO.StudentResult studentResult : scores.studentResults) {
+            // ToDo Support Team Exercises
+            List<StudentParticipation> participationsOfStudent = studentParticipations.stream()
+                    .filter(studentParticipation -> studentParticipation.getStudent().get().getId() == studentResult.id).collect(Collectors.toList());
+
+            for (StudentParticipation studentParticipation : participationsOfStudent) {
+                Exercise exercise = studentParticipation.getExercise();
+
+                Optional<Result> relevantResult = getRelevantResult(studentParticipation);
+
+                if (relevantResult.isPresent()) {
+                    Result result = relevantResult.get();
+                    Double achievedPoints = round((result.getScore() / 100.0 * exercise.getMaxScore()), 2);
+
+                    if (studentResult.overallPointsAchieved == null) {
```

Comment:
Why not initialize studentResult.overallPointsAchieved with 0. Then you don't need the conditional.

Reply:
makes more sense